tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -174,103 +209,118 @@ def _install_agent_on_node(self) -> None:
-                log.info("")
+                agent_test_logger.info("")
+                agent_test_logger.info("**************************************** %s ****************************************", self.context.suite_name)
+                agent_test_logger.info("")
 
                 results: List[str] = []
 
                 for test in test_suite:
-                    try:
-                        log.info("******************** [%s]", test.__name__)
-                        log.info("")
+                    result: str = "[UNKNOWN]"
+                    test_full_name = f"{self.context.suite_name} {test.__name__}"
+                    agent_test_logger.info("******** Executing %s", test_full_name)
+                    self._log.info("******** Executing %s", test_full_name)
+                    agent_test_logger.info("")
 
+                    try:
                         test(self.context).run()
-
-                        result = f"[Passed] {test.__name__}"
-
-                        log.info("")
-                        log.info("******************** %s", result)
-                        log.info("")
-
-                        results.append(result)
+                        result = f"[Passed] {test_full_name}"
+                    except AssertionError as e:
+                        failed.append(test.__name__)
+                        result = f"[Failed] {test_full_name}"
+                        self._log.error("%s", e)
+                        agent_test_logger.error("%s", e)
+                        self._log.error("%s", e)
                     except:  # pylint: disable=bare-except
-                        result = f"[Failed] {test.__name__}"
-
-                        log.info("")
-                        log.exception("******************** %s\n", result)
-                        log.info("")
-
-                        results.append(result)
                         failed.append(test.__name__)
-
-                log.info("**************************************** [Test Results] ****************************************")
-                log.info("")
+                        result = f"[Error] {test_full_name}"
+                        agent_test_logger.exception("UNHANDLED EXCEPTION IN %s", test_full_name)
+                        self._log.exception("UNHANDLED EXCEPTION IN %s", test_full_name)
+
+                    agent_test_logger.info("******** %s", result)
+                    agent_test_logger.info("")
+                    self._log.info("******** %s", result)
+                    results.append(result)
+
+                agent_test_logger.info("")
+                agent_test_logger.info("********* [Test Results]")
+                agent_test_logger.info("")
                 for r in results:
-                    log.info("\t%s", r)
-                log.info("")
+                    agent_test_logger.info("\t%s", r)
```

Comment:
why not added to main log? we know the summary in the main log too?

Reply:
it doesn't help there. it's a multi-line report and the output from different threads makes it hard to read, plus it won't always be at the end of the log, it'll be lost with the execution for other distros.

while executing the test suite, the purpose of the main log is just to show the progress of the test suite run. i duplicated test errors in the main log because that may be more convenient when checking the overall status of the test run.